packages/protocol/test-sol/common/FeeHandler.t.sol

```diff
@@ -0,0 +1,906 @@
+    _;
+  }
+
+  modifier setMaxSlippage() {
+    feeHandler.setMaxSplippage(address(stableToken), FIXED1);
+    _;
+  }
+
+  function test_CantDistributeWhenNotActive() public setUpBeneficiary {
+    vm.expectRevert("Handler has to be set to sell token");
+    feeHandler.distribute(address(stableToken));
+  }
+
+  function test_CantDistributeWhenFrozen() public setUpBeneficiary activateToken {
+    freezer.freeze(address(feeHandler));
+    vm.expectRevert("can't call when contract is frozen");
+    feeHandler.distribute(address(stableToken));
+  }
+
+  function test_DoesntDistributeWhenBalanceIsZero()
+    public
+    setBurnFraction
+    setMaxSlippage
+    setUpBeneficiary
+    activateToken
+    fundFeeHandlerStable(1e18)
+  {
+    // If we uncomment this the test should fail
+    // feeHandler.sell(address(stableToken));
+    vm.recordLogs();
+    feeHandler.distribute(address(stableToken));
+    Vm.Log[] memory entries = vm.getRecordedLogs();
+    assertEq(entries.length, 0);
+  }
+
+  function test_Distribute()
+    public
+    setBurnFraction
+    setMaxSlippage
+    setUpBeneficiary
+    activateToken
+    fundFeeHandlerStable(1e18)
+  {
+    feeHandler.sell(address(stableToken));
+
+    feeHandler.distribute(address(stableToken));
+
+    assertEq(stableToken.balanceOf(address(feeHandler)), 0);
+    assertEq(stableToken.balanceOf(EXAMPLE_BENEFICIARY_ADDRESS), 2e17);
+  }
+}
+
+contract FeeHandlerBurnCelo is FeeHandlerFoundry {
+  modifier activateToken() {
+    feeHandler.activateToken(address(celoToken)); // celoToken doesn't need to be added before activating
+    _;
+  }
+
+  modifier setBurnFraction() {
+    feeHandler.setBurnFraction(FixidityLib.newFixedFraction(80, 100).unwrap());
+    _;
+  }
+
+  modifier fundFeeHandler() {
+    uint256 celoAmount = 1e18;
+    celoToken.transfer(address(feeHandler), celoAmount);
+    _;
+  }
+
+  function test_BurnsCorrectly() public activateToken setBurnFraction fundFeeHandler {
+    feeHandler.burnCelo();
+    assertEq(celoToken.balanceOf(address(feeHandler)), 2e17);
+    assertEq(celoToken.getBurnedAmount(), 8e17);
+  }
+
+  function test_DoesntBurnPendingDistribution()
+    public
+    activateToken
+    setBurnFraction
+    fundFeeHandler
+  {
+    feeHandler.burnCelo();
+    assertEq(celoToken.getBurnedAmount(), 8e17);
+    // this is the amount pending distribution
+    assertEq(celoToken.balanceOf(address(feeHandler)), 2e17);
+
+    feeHandler.burnCelo();
+    assertEq(celoToken.getBurnedAmount(), 8e17);
+    // amount pending distribution should not be changed by second burn
+    assertEq(celoToken.balanceOf(address(feeHandler)), 2e17);
+  }
+
+  function test_DistributesCorrectlyAfterBurn()
+    public
+    activateToken
+    setBurnFraction
+    fundFeeHandler
+  {
+    feeHandler.burnCelo();
+    assertEq(celoToken.balanceOf(address(feeHandler)), 2e17);
+
+    feeHandler.setFeeBeneficiary(EXAMPLE_BENEFICIARY_ADDRESS);
+    feeHandler.distribute(address(celoToken));
```

Comment:
I see that in the original test, `feehandler.distribute()` was aslo being called with the stabletoken address. Any particular reason this changed?